build_tools/scripts/git/check_submodule_init.py

```diff
@@ -20,13 +29,28 @@ def run():
 
   output = os.popen("git submodule status")
   submodules = output.readlines()
+  runtime_submodules = [
+      "third_party/benchmark", "third_party/cpuinfo", "third_party/flatcc",
+      "third_party/googletest", "third_party/liburing", "third_party/libyaml",
+      "third_party/musl", "third_party/spirv_cross",
+      "third_party/spirv_headers", "third_party/tracy",
+      "third_party/vulkan_headers", "third_party/vulkan_memory_allocator",
+      "third_party/webgpu-headers"
+  ]
 
   for submodule in submodules:
     if (submodule.strip()[0] == "-"):
-      print(
-          "The git submodule '%s' is not initialized. Please run `git submodule update --init`"
-          % (submodule.split()[1]))
-      sys.exit(1)
+      if (args.runtime_only and submodule.split()[1] in runtime_submodules):
+        print(
+            "The git submodule '%s' is not initialized. "
+            "Please run `./build_tools/scripts/git/update_runtime_submodules.sh`"
+            % (submodule.split()[1]))
+        sys.exit(1)
+      elif (not args.runtime_only):
+        print(
+            "The git submodule '%s' is not initialized. Please run `git submodule update --init`"
+            % (submodule.split()[1]))
+        sys.exit(1)
```

Comment:
Let's leave this as an advanced functionality to start out with. I would still just print the generic message regardless

Reply:
Update the conditions to share the same error message.